docs: update supported Python versions and add onboarding steps #3441 issue solved

paras-chinchalkar opened the pull request:
Title / Commit: docs: update supported Python versions and add onboarding steps (#3441) — committed to the repo.

Files changed: [README.md](vscode-file://vscode-app/c:/Users/PARAS/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) (path: [README.md](vscode-file://vscode-app/c:/Users/PARAS/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html))
Replaced the hard-coded "Python3.7 through Python3.11" text with guidance that Augur aims to support current officially-supported Python releases (centered on Python 3.11).
Added cross-platform virtualenv examples (PowerShell for Windows and bash for macOS/Linux).
Added an Onboarding for New Contributors section with steps: join Slack, read CONTRIBUTING.md, run Augur, find starter issues, and ask for help.

sgoggins commented:
@MoralCode : I added a "discussion" label here because I'm not certain ... BUT, I think we have other open PR's that address the python versioning issues.

MoralCode commented:
> Title / Commit: docs: update supported Python versions and add onboarding steps (#3441) — committed to the repo.

Please make the titles of your Pull requests more descriptive of the actual issue being solved

paras-chinchalkar commented:
> > Title / Commit: docs: update supported Python versions and add onboarding steps (#3441) — committed to the repo.
> 
> Please make the titles of your Pull requests more descriptive of the actual issue being solved



Problem 1: Outdated Python version documentation
Issue: Hard-coded "Python3.7 through Python3.11" becomes outdated, and OSX-only examples don't help Windows/Linux users.
Solution: Replaced with flexible language about supporting current Python versions (centered on 3.11) and added cross-platform virtual environment examples for Windows (PowerShell) and macOS/Linux.
Impact: Documentation stays current longer and helps contributors on all platforms.
Problem 2: Missing onboarding guidance
Issue: New contributors lacked a clear starting point; onboarding info was scattered.
Solution: Added an "Onboarding for New Contributors" section in README.md with steps: join Slack, read CONTRIBUTING.md, run Augur locally, find starter issues (first-timers-only or good first issue), and ask for help.
Impact: Clear, visible path for newcomers to start contributing.
Problem 3: Low discoverability
Issue: Critical onboarding info was buried in docs.
Solution: Placed onboarding steps directly in README.md for immediate visibility.
Impact: Easier for new contributors to find and follow the process.
Issues resolved: #3441, #3448

MoralCode commented on README.md:
```diff
@@ -43,11 +43,26 @@ For more information on [how to get involved on the CHAOSS website](https://chao
+After activating the virtual environment, install the project's development dependencies as described in the developer docs.
+
+For full, up-to-date setup instructions (including Docker-based development), see the developer documentation: https://oss-augur.readthedocs.io/en/main/development-guide/toc.html
+
```
I like the way this starts out in terms of talking about the python support, but I also liked @cnaples79's version (#3443) which says explicitly that augur uses UV for managing python dependencies.

Uv as a dependency manager doesnt really have a specific "activation" process, you just run commands through `uv run` and it handles it all for you, including downloading and setting up the virtual env behind the scenes.

paras-chinchalkar commented on README.md:
```diff
@@ -76,6 +91,17 @@ but we are actively adding to it all the time.
+- **Find a starter task**: Look for issues labeled `first-timers-only` or `good first issue` to make your first contribution.
+- **Ask for help**: If you get stuck, ask on Slack, comment on your issue/PR, or bring questions to community meetings — maintainers are happy to assist.
+
+
```
sure
